wordpress-mobile/WordPress-iOS
# Feature/plugin vc network

This PR attempts to make the plugin details view aware of the network status, and reload if necessary.

There is a lot I don't like about this implementation, and I am opening a PR mostly to gather feedback from you @etoledom , and explore any other alternatives you might suggest.

I did not implement the logic to observe changes in the network status in the view model because I af afraid to break something else that I might not be aware of.

To test:
Scenario #1 
1. Navigate a to a plugin details view. with an active network connection. 
2. Enter flight mode.
3. There should be an alert like the following (the screenshot is the Spanish localisation)
![img_2919](https://user-images.githubusercontent.com/2722505/43387661-99547d2a-9419-11e8-9d24-1a9bb141a311.PNG)
4. Exit flight mode, the view should not reload

Scenario #2
1. Navigate a to a plugin details view. with an inactive network connection. 
2. There should be an error message like the following.
![img_2920](https://user-images.githubusercontent.com/2722505/43387688-aea922e8-9419-11e8-98e9-7a217dfdbfb2.PNG)
3. Exit flight mode
4. The view should reload and display the expected content
![img_2921](https://user-images.githubusercontent.com/2722505/43387704-b911210e-9419-11e8-8fda-e7ba93ff980e.PNG)
